lib/datadog/profiling/component.rb

```diff
@@ -85,6 +85,7 @@ def self.build_profiler_component(settings:, agent_settings:, optional_tracer:)
           cpu_time_enabled: RUBY_PLATFORM.include?('linux'), # Only supported on Linux currently
           alloc_samples_enabled: allocation_profiling_enabled,
           heap_samples_enabled: heap_profiling_enabled,
+          heap_size_enabled: heap_profiling_enabled,
```

Comment:
Note to our future selves, a separate config for this gets introduced in https://github.com/DataDog/dd-trace-rb/pull/3360 .